### PR TITLE
feat(FOROME-1326): add an ability to delete and modify drawer presets

### PR DESCRIPTION
**Pull request by Lapk1n**
https://quantori.atlassian.net/browse/FOROME-1326

### CODE REVIEW

File: src/pages/ws/ui/variant-drawer/variant-drawer-header/drawer-layout-control/drawer-layout-control.tsx [hunk 1]
```diff
@@ -1,48 +1,92 @@
...
 import { SavePresetDialog } from './save-preset-dialog'
 
 interface IDrawerLayoutControlProps {
-  className?: string
   layoutMode: VariantDrawerLayoutMode
-  onChangeLayoutMode: (mode: VariantDrawerLayoutMode) => void
   gridPresets: IVariantDrawerGridPreset[]
+  windowsOpenState: boolean
+  appliedPreset: string | null
+  className?: string
```
Comment (QSdmitrioul) [inline src/pages/ws/ui/variant-drawer/variant-drawer-header/drawer-layout-control/drawer-layout-control.tsx]: Argument

File: src/pages/ws/ui/variant-drawer/variant-drawer-header/drawer-layout-control/drawer-layout-control.tsx [hunk 2]
```diff
@@ -1,48 +1,92 @@
...
+
+  const { popoverAnchor, isPopoverOpen, onToggle, closePopover } = usePopover()
+
+  const [selectedPreset, setSelectedPreset] = useState<string>('')
+
+  useEffect(() => {
+    if (isPopoverOpen && selectedPreset !== appliedPreset) {
+      setSelectedPreset(appliedPreset || '')
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (QSdmitrioul) [inline src/pages/ws/ui/variant-drawer/variant-drawer-header/drawer-layout-control/drawer-layout-control.tsx:61]: Why you didn't add other dependencies?

File: src/pages/ws/ui/variant-drawer/variant-drawer-header/drawer-layout-control/drawer-layout-control.tsx [hunk 3]
```diff
@@ -59,58 +103,77 @@ export const DrawerLayoutControl = ({
...
-          onSave={presetName => {
-            onSaveGridPreset(presetName)
-            closeSavePresetDialog()
-          }}
+          onSave={handleSavePreset}
           presets={gridPresets}
         />
       )}
+
+      <SolutionChangeDialog
```
Comment (AndreyUstyumenko) [inline src/pages/ws/ui/variant-drawer/variant-drawer-header/drawer-layout-control/drawer-layout-control.tsx]: I don't think it's a proper component to use here

File: src/pages/ws/ui/variant-drawer/variant-drawer-header/drawer-layout-control/drawer-layout-control.tsx (hunk 2 above)
Comment (Lapk1n) [reply src/pages/ws/ui/variant-drawer/variant-drawer-header/drawer-layout-control/drawer-layout-control.tsx:61]: Cause they will clear the selected preset on every select

**Review by QSdmitrioul — APPROVED** [review]

**Review by EvgeniySobolev — APPROVED** [review]

**Review by AndreyUstyumenko — APPROVED** [review]

**Pull request merged**